### PR TITLE
fix(rep-widget): Reload reps when a new wallet is imported

### LINKED ISSUE #150: Current rep in the sidebar not updated when importing Ledger wallet
If having another account in Nault and adding a Ledger, the rep stays for the old wallet until you click on it which takes you to the rep page and refreshes. Not sure exactly where this fault is. Probably some "isLedger" variable that has been missed.

### PR DESCRIPTION
**Pull request by Joohansson**
-Subscribe to when a new wallet is created
-Reload reps accordingly

Fixes #150 

Is this a valid approach, or can it be made more simple? Tested and working for new seed, ledger or private key import. See the linked issue.

### CODE REVIEW

**Review by keerifox — CHANGES_REQUESTED** [review]
https://github.com/keerifox/Nault/commit/d132205b11559bbb9ea9b0c015f23b3093aa165f fixes reps not updating for new wallets (not ledger / seed imports but generated ones)

https://github.com/keerifox/Nault/commit/23dff62eb6dc6ac08256b51c3c70fb458b75e870 hides reps even before all the api calls for new reps are completed, also reduces calls to getRepresentativesOverview() which was called an extra time when detecting changeable reps

**Review by keerifox — APPROVED** [review]

File: src/app/components/change-rep-widget/change-rep-widget.component.ts [hunk 1]
```diff
@@ -37,7 +39,19 @@ export class ChangeRepWidgetComponent implements OnInit {
       this.updateDisplayedRepresentatives();
     });
 
-    await this.repService.detectChangeableReps();
+    // Detect if a wallet is reset
+    this.walletService.wallet.newWallet$.subscribe(async bool => {
+      if (bool) {
```
Comment (BitDesert) [inline src/app/components/change-rep-widget/change-rep-widget.component.ts]: Please change `bool` to a more descriptive name like `shouldReload` or sth. like that.

File: src/app/components/change-rep-widget/change-rep-widget.component.ts [hunk 2]
```diff
@@ -37,7 +39,19 @@ export class ChangeRepWidgetComponent implements OnInit {
...
+        console.log('Reloading representatives..');
+        this.selectedAccount = null;
+        this.representatives = [];
+        this.changeableRepresentatives = [];
+        this.showRepChangeRequired = false;
+        this.updateDisplayedRepresentatives();
+        this.representatives = await this.repService.getRepresentativesOverview(); // calls walletReps$.next
```
Comment (BitDesert) [inline src/app/components/change-rep-widget/change-rep-widget.component.ts]: This could get moved to a new function like `resetRepresentatives` or so.

**Review by BitDesert — CHANGES_REQUESTED** [review]

File: src/app/components/change-rep-widget/change-rep-widget.component.ts (hunk 1 above)
Comment (Joohansson) [reply src/app/components/change-rep-widget/change-rep-widget.component.ts]: done

File: src/app/components/change-rep-widget/change-rep-widget.component.ts (hunk 2 above)
Comment (Joohansson) [reply src/app/components/change-rep-widget/change-rep-widget.component.ts]: done

**Review by BitDesert — APPROVED** [review]

**Pull request merged**